### PR TITLE
Feature/first version

### PR DESCRIPTION
**Pull request by edersonbrilhante**
## Description
This PR will create the first version of the wrapper

### Running locally
```
docker-compose -f docker/docker-compose.yaml up
```
#### Running using curl
```
curl --location --request POST 'http://localhost:8080/analysis' \
--header 'Content-Type: application/json' \
--data-raw '
{"image":"alpine:3.1"}'
```
#### Result expected
```
{
    "error": "",
    "result": [
        {
            "Target": "alpine:3.1 (alpine 3.1.4)",
            "Type": "alpine",
            "Vulnerabilities": null
        }
    ],
    "success": true
}
```

### CODE REVIEW

**Review by szymonwyrwiak — COMMENTED** [review]
Was wondering about adding 

```
# ====================================================================
# See https://pre-commit.com              for more information
# See https://pre-commit.com/hooks.html   for more hooks
# ====================================================================
```
to the top.

**Review by szymonwyrwiak — COMMENTED** [review]
go-lint needs separate installation through 
`go get -u golang.org/x/lint/golint`
without it
`pre-commit run --all-files` 
will fail with message like
`go lint..................................................................Failed
hook id: go-lint
exit code: 1
/Users/user/.cache/pre-commit/repoetk6vpau/run-go-lint.sh: line 7: golint: command not found`

**Review by szymonwyrwiak — COMMENTED** [review]
We have to add informations about:

- Usage
-- How to run server
-- How to consume APIs

later one

**Review by szymonwyrwiak — CHANGES_REQUESTED** [review]

File: api/analysis/analysis.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	wtypes "github.com/william-Hill-Online/trivy-wrapper-api/api/types"
+)
+
+const logCreateScanRequest = "analysis.RunAnalysis"
+
+// RunAnalysis is the function to execute trivy client and return the results.
+func RunAnalysis(analysis wtypes.AnalysisRequest) (wtypes.AnalysisResult, error) {
+
+	remoteTrivy := os.Getenv("REMOTE_TRIVY")
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply api/analysis/analysis.go:19]: `REMOTE_TRIVY` is the IP + port running the `trivy_server`. If you change the IP and port of `trivy_server`, you have to change the value in ENVVAR `REMOTE_TRIVY` with the new IP and port.

File: api/routes/healthcheck.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2019 Globo.com authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply api/routes/healthcheck.go:3]: Good catch. I copied the license block as example and I forgot to change :)

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module github.com/william-Hill-Online/trivy-wrapper-api
+
+go 1.14
+
+require (
+	github.com/aquasecurity/fanal v0.0.0-20200901125602-1b5f79c38073
+	github.com/aquasecurity/trivy v0.11.0
+	github.com/labstack/echo/v4 v4.1.17
+	github.com/sirupsen/logrus v1.6.0
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply go.mod:9]: This is normal. Each dep in go.mod has more deps, so go.sum list all deps needed. You will have the same in projects in python or javascript

File: README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+## trivy wrapper API
+This project works as a wrapper api for [trivy](https://github.com/aquasecurity/trivy) client
+
+Run docker 
+```
+docker-compose -f docker/docker-compose.yaml up
+```
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply README.md:7]: Do you think the curl in the description in this MR could be enough?

File: docker/docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        - "../api:/go/src/github.com/William-Hill-Online/trivy-wrapper-api/api/"
+    ports:
+      - "8080:8080"
+    networks:
+        - trivy_net
+    depends_on:
+        - trivy-server
+
+networks:
+    trivy_net:
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply docker/docker-compose.yaml:31]: This docker-compose is just to help the development. So the server and the API can be started easily for development purposes

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+repos:
```
[the remark this answers is not in the document]
Comment (edersonbrilhante) [reply .pre-commit-config.yaml:5]: Done

**Comment by edersonbrilhante** [comment]
> We should add test coverage later one.

Agreed.

**Review by szymonwyrwiak — APPROVED** [review]
LGTM

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (szymonwyrwiak) [reply README.md]: For mvp it's enough

**Pull request merged**